python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -312,6 +324,26 @@ def _report_current_usage(
         if self._global_info:
             self._global_info.set_description(resources_status)
 
+    def _update_dataset_metrics(self):
+        stats = {metric: 0 for metric in DataMetric}
+
+        for op in self._topology:
+            metrics = op.metrics
+            resource_usage = op.current_resource_usage()
+
+            stats[DataMetric.BYTES_SPILLED] += metrics.obj_store_mem_spilled
```

Comment:
Can we just reuse the same key as in OpRuntimeMetrics? Also, can we just report the entire metrics `metrics.as_dict()`?
I'm concerned about the maintenance overheads and inconsistency.

Reply:
There will be metrics that are not held in `OpRuntimeMetrics` right? what key would we use for those?  And are you saying we should remove the `DataMetric` enum as well?

Reply:
I think eventually we should move everything to `OpRuntimeMetrics`. and then `DataMetric` doesn't seem necessary anymore.

Reply:
actually, we can easily move `op.current_resource_usage()` to `OpRuntimeMetrics`. Just return incremental_resource_usage() * num_running_tasks

Reply:
Would we still want to store these keys as variables somewhere? Otherwise we would have to hardcode these keys in tests and in stats.py